### PR TITLE
IN-50 allow periodical actions +  implement engine fire

### PR DESCRIPTION
**Pull request by Dummyrunner**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -78,6 +79,11 @@ def process_keyboard_events(actions_while_key_pressed, actions_on_key_down):
             action.execute_action()
 
 
+def process_each_frame_events(actions_each_frame):
+    for act in actions_each_frame:
+        act.execute_action()
```
Comment (beneblubb) [inline main.py:84]: For getting a new state of an object, until now '.update' was used. Is it intentional to change this here?

File: main.py [hunk 2]
```diff
@@ -133,6 +149,26 @@ def main():
...
+
+    glue_engine_fire_down_to_ego = lambda: setattr(
+        ego_engine_fire,
+        "pos",
+        Vec2d(
+            ego.rect.midbottom[0],
+            ego.rect.midbottom[1]
+            + 0.5 * CommonConstants.EXHAUST_FIRE_DOWN_DIMENSIONS.y,
+        ),
+    )
```
Comment (beneblubb) [inline main.py:170]: As told via discord, as we probably will glue many objects to another, i would prefer a standardised way so we don't have to do it manually. For this, if we can't find a better solution, i would probably try to make this a functionality of the LandingGameObject.

**Review by beneblubb — APPROVED** [review]
Nice addition to the functionality of the game 👍

File: main.py (hunk 2 above)
Comment (Dummyrunner) [reply main.py:170]: I agree, seeing it as a topic out of scope for this PR.

**Pull request merged**